### PR TITLE
Adding AWS CloudTrail to swagger

### PR DESCRIPTION
**Pull request by mahmoudraedhasanmsft**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by mahmoudraedhasanmsft** [comment]
@shahabhijeet please review my PR. thanks

**Comment by loarabia** [comment]
@shahabhijeet this looks like it is getting stale is there a blocker?

**Comment by mahmoudraedhasanmsft** [comment]
Hi @loarabia , @shahabhijeet ,
Who can review this PR?
Thanks

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json [hunk 1]
```diff
@@ -2363,6 +2370,52 @@
...
+      "type": "object",
+      "description": "Represents Amazon Web Services CloudTrail data connector.",
+      "allOf": [
+          {
+              "$ref": "#/definitions/DataConnector"
+          }
+      ],
+      "properties": {
+          "properties": {
+              "x-ms-client-flatten": true,
```
Comment (shahabhijeet) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: @mahmoudraedhasanmsft please verify if flattening produces the model that you expect.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

**Comment by mahmoudraedhasanmsft** [comment]
@shahabhijeet I think it should be the same like the other connectors.
@nirgfani , what do you think?

**Comment by mahmoudraedhasanmsft** [comment]
@shahabhijeet , I checked it with my team and it's ok. Please approve if you see that everything is OK.

**Comment by mahmoudraedhasanmsft** [comment]
@loarabia can you please approve the PR?

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**